### ISSUE TITLE
jQuery/javascript from other elements

### ISSUE THREAD

**Issue by Delpierooo123** [luruke/barba.js#135]
Hi.

We are using your barba.js script in our Wordpress theme. On some pages we use differrent elements (using shorcodes) - for example an owl slider. That slider works well, if we open the page directly. But if we navigate to that page from another (for example opening home and clicking about where the slider is), the slider is not working. Where should we put such jQuery/javascript?

We use FadeTransition example from here: http://barbajs.org/transition.html (a bit modified).

Also, will other jQuery/javascript used have to be entered aslo to a special place? For example if some  plugin is used.

EDIT: Have found a solution with http://barbajs.org/views.html. 

But what if a plugin that has its own jQuery is used?

Best regards,

Delpierooo

**Comment by luruke** [comment]
Hello @Delpierooo123 

You should "run/start" your jQuery plugin at page change, for example:
```
Barba.Dispatcher.on('newPageReady', function() {
  // check if there is a new owl slider
  // init new slider
});
```

**Comment by Delpierooo123** [comment]
Hi. 

Will try, thanks.

What about jQuery plugin that someone will use and I can not predict?

Best regards,

Delpierooo

**Comment by Delpierooo123** [comment]
Hi.

one more question/issue I am facing:

Can I somehow call the " fadeIn: function() {.." when a page is reloaded as well? For example I have different animations done isnide this function. Let say for about page. But if I refresh it or access it via direct link, that animations do not trigger.

How can I call this function on each refresh/reload?


Many thanks.

**Comment by luruke** [comment]
@Delpierooo123 You have different ways to do this.

- add this animation/onboarding inside your View (onEnterCompleted) related to that page.
- use `newPageReady` event and check that the [history](http://barbajs.org/docs/Barba.HistoryManager.html#.history) length is 1 - so you can run the animation if the user is landing directly on a specific page.

**Comment by Delpierooo123** [comment]
Hi,

can you pelase help me a bid more with the code?

Here is our example: http://pegodesign.com/wp-themes/opty/ 

For example, if you click About the transition is ok. But if I refresh the about page the fadeIn function does not get "read" - sidebar hover effect is included in there for example.

How to I achive the same animations that I have in fadeIn function?


Thanks in advance.

**Comment by grahamd711** [comment]
+1 

Having an issue with this as well. I've wrapped my jquery plugin codes in an initialization function. I've called this on newPageReady and it has a slight delay waiting for the document to load before the Jquery code runs.

I've also tried adding the initialization function to each view specific to the page the animation runs, but run into issues when I navigate back to the parent page because the Jquery is waiting on the document ready.

I'm using a fullpage.js which needs to initialize on each page of the website in order for the 'full page' aspect to work. Therefore, what is the best way to initialize certain Jquery plugins that need to be used on each page? I have seen #118 and #126 but still seem to be stuck.

Thanks!
